storage/uri.go

```diff
@@ -59,53 +61,131 @@ func (u *uri) String() string {
 	return u.raw
 }
 
+// parentGeneric is a generic function that returns the last element of a
+// path after splitting it on "/". It should be suitable for most URIs.
+func parentGeneric(location string) (string, error) {
```

Comment:
It seems like this has not got tests because we're testing only the file version

Reply:
I added multiple tests for this functions

Reply:
Yeah, I never got around to testing it explicitly after I switched to using `filepath`. Thank you @PucklaMotzer09